stdlib/public/core/StringCreate.swift

```diff
@@ -15,12 +15,30 @@
 internal func _allASCII(_ input: UnsafeBufferPointer<UInt8>) -> Bool {
   // NOTE: Avoiding for-in syntax to avoid bounds checks
   //
-  // TODO(String performance): Vectorize and/or incorporate into validity
-  // checking, perhaps both.
+  // TODO(String performance): SIMD-ize
   //
   let ptr = input.baseAddress._unsafelyUnwrappedUnchecked
   var i = 0
-  while i < input.count {
+
+  let count = input.count
+  let stride = MemoryLayout<UInt>.stride
+  let address = Int(bitPattern: ptr)
+
+  while (address &+ i) % stride != 0 && i < count {
+    guard ptr[i] <= 0x7F else { return false }
```

Comment:
This isn't really semantically important, but just for readability could we stick to (& 0x80…) rather than using that in one place and <= 0x7F in the other?

Reply:
Makes sense

Reply:
Depending on what the expected trip-count is here, it may not be worth explicitly aligning. Did you experiment with this at all?

Reply:
No, and there's not really any time for much experimentation if this is going to ship. According to @jckarter, our loads and stores assumed aligned pointers, so I thought this was necessary for correctness.

Reply:
Good enough for now, then.